content/articles/visualizing-repositories-using-pygal/index.md

```diff
@@ -0,0 +1,227 @@
+
+![Python Works with the Most Stars on GitHub](/engineering-education/visualizing-repositories-using-pygal/visualizing.jpeg)
+
+### Refining Pygal Charts
+Let's fine-tune the look of our graph. First, reorganize the code somewhat by establishing a configuration object that includes all of our adjustments to pass to `Bar()`:
+
+```python
+import requests
+import pygal
+from pygal.style import LightColorizedStyle as LCSe, LightenStyle as LSe
+#  Start an API request.
+url = 'https://api.github.com/search/repositories?q=language:python&sort=star'
+request = requests.get(url)
+print("Status  code:", request.status_code)
+# In a variable, save the API response.
+response_dictionary =request.json()
+print("Total-repos:", response_dictionary['total_count'])
+# Investigate the repository's contents.
+repos_dicts = response_dictionary['items']
+names, stars = [], []
+for repos_dict in repos_dicts:
+    names.append(repos_dict['name'])
+    stars.append(repos_dict['stargazers_count'])
+# Construct a visual picture
+our_style = LSe('#333366', base_style=LCSe)
+myconfig = pygal.Config()   # make an instance of Pygal’s Config class
+myconfig.x_label_rotation = 45    # set the  attribute x_label_rotation to 45
+myconfig.show_legend = False    #set the attribute show_legend to false
+myconfig.title_font_size = 23   #choose the text size for the heading of the chart
+myconfig.label_font_size = 13   # choose the text size for the minor label (The minor labels are the project names along the x-axis)   
+myconfig.major_label_font_size = 17 #set the text size for the major label(The major labels are just the labels on the y-axis)
+myconfig.truncate_label = 15  #To reduce the length of long project names to 15 characters, employ truncate label.
+myconfig.show_y_guides = False  #set show_y_guides to False hide the horizontal lines on the graph
+myconfig.width = 1000   #Choose a custom width for the graph so that it takes up maximum browser area
+chart = pygal.Bar(myconfig, style=our_style)  #make an instance of Bar, and pass myconfig and style as arguments, respectively
+chart.title = 'Python Projects with the Most Stars on GitHub'
+chart.x_labels = names
+chart.add('', stars)
+chart.render_to_file('py_repos.svg')
+```
+
+The restyled chart is shown below:
+
+![The styling for the chart has been refined.](/engineering-education/visualizing-repositories-using-pygal/refining.jpeg)
+
+### Customizing Tooltips
+Floating a mouse above a single bar displays data it represents. This is known as a tooltip, and it usually displays the tally of stars that a project possesses. Let's make a custom tooltip to display the descriptions of each project.
+
+Let's have a look at a simple example that uses the first three projects plotted singly with custom labels for each bar. Rather than passing a list of values to `add()`, we'll pass a list of dictionaries with the code below:
```

Comment:
```suggestion
Let's have a look at an example that uses the first three projects plotted singly with custom labels for each bar. Rather than passing a list of values to `add()`, we'll pass a list of dictionaries with the code below:
```